metis-common/src/main/java/eu/europeana/metis/zoho/ZohoAccessClient.java

```diff
@@ -53,27 +47,31 @@ public ZohoAccessClient() {
    * tokens from the persistence storage.
    * </p>
    *
-   * @param grantToken the grant token key to be used
+   * @param grantToken the grant token key to be used. Can be null
+   * @param zcrmConfigurations the key value map of zoho configuration
    * @throws ZohoOAuthException if initialization of {@link ZCRMRestClient} failed.
    */
-  public ZohoAccessClient(String grantToken) throws ZohoOAuthException {
+  public ZohoAccessClient(@Nullable String grantToken, HashMap<String, String> zcrmConfigurations)
```

Comment:
SonarLint warning: Use Map<String, String> instead of HashMap to define input type.

Reply:
Yes I had the same. The issue comes from 
ZCRMRestClient.initialize() which requires a HashMap apparently.
I could change the parameter to Map, but then I have to cast it to HashMap which can be unsafe.
What do you think?

Reply:
Yes I am aware of this. The reason is that the ZCRMRestClient.initialize(zcrmConfigurations); accepts a HashMap, so I thought the class here should also be a HashMap.
The alternative is to provide a Map here and cast it before sending it to the inner class..
Not sure what is better, somehow the user of this class should know that the Map has to be of type HashMap. Perhaps though documentation and we check the type of the Map inside here with an exception if it's not the type we want?
What do you think?

Reply:
OK, so we specifically need it to be a HashMap here? Then I see two options that I don't dislike: 
1. Make it a generic map and then use `new HashMap<>(zcrmConfigurations)` that we then send to zoho. That way we accept generic input and still satisfy Zoho.
2. Put an exception for the sonar lint rule in the code with a piece of comments explaining this.
I think I slightly favor the first one. What do you think?

Reply:
Yeah number 1 seems better